### PR TITLE
Sscssi 25

### PR DESCRIPTION
**Pull request by Andrew-McMahon7**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/SSCSSI-25

### Change description ###
Update CitizenLoginService to use UpdateCcdCaseService UpdateCaseV2

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/service/CitizenLoginService.java [hunk 1]
```diff
@@ -164,8 +167,25 @@ public void findAndUpdateCaseLastLoggedIntoMya(IdamTokens citizenIdamTokens, Str
     }
 
     private void updateCaseWithLastLoggedIntoMya(String email, SscsCaseDetails caseByAppealNumber) {
-        updateSubscriptionWithLastLoggedIntoMya(caseByAppealNumber, email);
-        ccdService.updateCase(caseByAppealNumber.getData(), caseByAppealNumber.getId(), EventType.UPDATE_CASE_ONLY.getCcdType(), "SSCS - update last logged in MYA", UPDATED_SSCS, idamService.getIdamTokens());
+        if (citizenLogicServiceV2) {
+            updateCcdCaseService.updateCaseV2(
```
Comment (nitinprabhuhmcts) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/CitizenLoginService.java:171]: is there too much spacing here ?

**Review by alivenichoppa — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/CitizenLoginService.java [hunk 2]
```diff
@@ -14,15 +14,13 @@
...
 import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.stereotype.Service;
 import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
-import uk.gov.hmcts.reform.sscs.ccd.domain.CcdValue;
-import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
-import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseDetails;
-import uk.gov.hmcts.reform.sscs.ccd.domain.Subscription;
-import uk.gov.hmcts.reform.sscs.ccd.domain.Subscriptions;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
```
Comment (nitinprabhuhmcts) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/CitizenLoginService.java:20]: we should avoid * imports and just include one needed unless there are quite large to be imported

File: src/main/java/uk/gov/hmcts/reform/sscs/service/CitizenLoginService.java [hunk 3]
```diff
@@ -164,8 +167,15 @@ public void findAndUpdateCaseLastLoggedIntoMya(IdamTokens citizenIdamTokens, Str
     }
 
     private void updateCaseWithLastLoggedIntoMya(String email, SscsCaseDetails caseByAppealNumber) {
-        updateSubscriptionWithLastLoggedIntoMya(caseByAppealNumber, email);
-        ccdService.updateCase(caseByAppealNumber.getData(), caseByAppealNumber.getId(), EventType.UPDATE_CASE_ONLY.getCcdType(), "SSCS - update last logged in MYA", UPDATED_SSCS, idamService.getIdamTokens());
+        if (citizenLogicServiceV2Enabled) {
+            updateCcdCaseService.updateCaseV2(caseByAppealNumber.getId(), EventType.UPDATE_CASE_ONLY.getCcdType(), "SSCS - update last logged in MYA",
```
Comment (nitinprabhuhmcts) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/CitizenLoginService.java:171]: adding a log would be nice

**Review by ruban72 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/reform/sscs/service/AbstractCitizenLoginServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,472 @@
+package uk.gov.hmcts.reform.sscs.service;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
```
Comment (nitinprabhuhmcts) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/AbstractCitizenLoginServiceTest.java:8]: * imports not recommended unless there are lot of classes from that package

File: src/test/java/uk/gov/hmcts/reform/sscs/service/AbstractCitizenLoginServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,472 @@
...
+import static uk.gov.hmcts.reform.sscs.util.DataFixtures.someOnlineHearing;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
```
Comment (nitinprabhuhmcts) [inline src/test/java/uk/gov/hmcts/reform/sscs/service/AbstractCitizenLoginServiceTest.java:18]: same here

**Review by alivenichoppa — APPROVED** [review]

**Review by nitinprabhuhmcts — APPROVED** [review]

**Review by alivenichoppa — APPROVED** [review]

**Review by alivenichoppa — APPROVED** [review]

**Pull request merged**